### PR TITLE
Update balance proof

### PR DESCRIPTION
**Pull request by palango**
This updates the BP to the new design. We discussed that the MS will only need a BP with the `balance_hash` set, while the PFS needs full information (`locked_amount` and `transferred_amount`)

Right now the BP-message handles both cases and might include the extra data. I'm not sure if this is the best design.

We also have to update the `MonitorRequest` @pcppcp

### CODE REVIEW

**Comment by palango** [comment]
Yes, please do that.

Also note that the signing stuff is copied from here: https://github.com/raiden-network/raiden-contracts/pull/39/files#diff-a8f4ba6a11d8d896b2d05b76786b76d7

File: raiden_libs/messages/balance_proof.py [hunk 1]
```diff
@@ -1,85 +1,112 @@
...
         self.nonce = nonce
-        self.locksroot = locksroot
-        self.transferred_amount = transferred_amount
-        self.extra_hash = extra_hash
-        self.signature = signature
         self.chain_id = chain_id
+        self.signature = signature
+
+        if transferred_amount and locked_amount and locksroot:
+            assert self.hash_balance_data(
```
Comment (palango) [inline raiden_libs/messages/balance_proof.py]: Not sure if we need to do that here. Maybe we should add a `validate` method instead.

File: raiden_libs/messages/balance_proof.py [hunk 2]
```diff
@@ -1,85 +1,112 @@
...
+        ])
 
     @classmethod
     def deserialize(cls, data):
         jsonschema.validate(data, BALANCE_PROOF_SCHEMA)
-        ret = cls(
-            data['channel_id'],
-            data['contract_address'],
+        result = cls(
+            data['channel_identifier'],
```
Comment (palango) [inline raiden_libs/messages/balance_proof.py]: other required stuff should be added here

File: requirements_dev.txt [hunk 3]
```diff
@@ -14,4 +14,4 @@ pytest
 pytest-runner
 pdbpp
 
-git+https://github.com/raiden-network/raiden-contracts.git
+git+https://github.com/loredanacirstea/raiden-contracts.git@update-privacy
```
Comment (palango) [inline requirements_dev.txt]: Remove that once the branch is merged.

**Comment by palango** [comment]
Added some type annotations. remaining issues: 
```
raiden_libs/test/mocks/client.py:61: error: Need type annotation for 'client_registry'
raiden_libs/test/mocks/client.py:160: error: "None" has no attribute "functions"
raiden_libs/test/mocks/client.py:260: error: "BalanceProof" has no attribute "extra_hash"
```

File: raiden_libs/messages/monitor_request.py [hunk 4]
```diff
@@ -22,95 +20,70 @@ class MonitorRequest(Message):
...
     def __init__(
         self,
-        channel_identifier: ChannelIdentifier,        # uint256
-        nonce: int,                     # uint64
-        transferred_amount: int,        # uint256
-        locksroot: str = None,               # str -> bytes 32
-        extra_hash: bytes = None,         # bytes 32
-        signature: bytes = None,          # bytes
+        balance_proof: BalanceProof,
+        non_closing_signature: str = '',
```
Comment (palango) [inline raiden_libs/messages/monitor_request.py]: Should this be `None` by default?

Comment (pcppcp) [reply raiden_libs/messages/monitor_request.py]: yes none would be better, I'll update & rebase

**Pull request merged**